base_location/models/company.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Nicolas Bessi, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    city_id = fields.Many2one(
+        'res.city',
+        compute='_compute_address',
+        inverse='_inverse_city_id',
+        string="City"
+    )
+
+    zip_id = fields.Many2one(
+        'res.better.zip',
+        string='ZIP Location',
+        compute='_compute_address',
```

Comment:
Not sure what happens with GH, but I re-comment to get it displayed.

AFAIK `res.company` extends `res.partner`, so `zip_id` should be there out of the box, right?

Reply:
Yeah, definitively I don't understand why the rest of the stuff would be needed having this easy approach of simply adding the field to the view, as in previous versions.

Reply:
That is not true, as you can check on [odoo company definition](https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/res/res_company.py#L11).

Reply:
Well, indeed I was wrong: `res.company` does not extend `res.partner`. In any case, wouldn't it be enough to add the field as related instead of repeating the compute methods?

Reply:
@Yajo But odoo chose to duplicate the address fields into res.company. And we should respect that IMHO.

Reply:
Yes, but [look at how it does it: basically reading and writing the field from the linked partner](https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/res/res_company.py#L101-L130). That's basically a more complex way of using a related field, isn't it?

Reply:
@Yajo I propose that we stick to the way odoo works now. And maybe we can propose to odoo to change their approach upstream, and so when they fix in future versions we also fix.

Reply:
I don't agree. If there are no reasons for doing that way, we don't have to do the same as Odoo. Let's keep the code simple. The only possible reason I see if it's there's any issue when you first create a company. Please check that case. If not, KISS principle.

Reply:
@pedrobaeza I have been testing and it is proving a mess to mix the two methods. When you make the fields related, the constrains that we added in res.partner, https://github.com/OCA/partner-contact/pull/488/files#diff-2f72fbcaf854c88b3e175783b321cd16R64, will fail. because when you change the city_id in the company, you are effectively changing it in the partner. The constrains on the partner are evaluated before the inverse methods update the other fields (city, etc..).

And we need constrains in the partner to ensure consistency.

So, as a conclusion, address fields are very related to each other. Either you make them all related to the partner in company, or you don't for all of them. But mixing methods produces inconsistencies.

Reply:
Thanks for the explanation, Jordi. OK, that's a good reason for keeping things this way. That was the explanation I was looking for keeping things this way.

Reply:
Interesting. Maybe you could add a comment or known issue on that matter, to avoid future iterations on the same subject unless odoo changes its way...